src/test/java/BaseTest.java

```diff
@@ -1,11 +1,99 @@
 import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
 import org.openqa.selenium.chrome.ChromeOptions;
-import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.*;
+
+import java.time.Duration;
+import java.util.UUID;
+
 
 public class BaseTest {
 
+    public WebDriver driver;
+    public String url = "https://qa.koel.app/";
+
     @BeforeSuite
     static void setupClass() {
         WebDriverManager.chromedriver().setup();
     }
+    @BeforeMethod
+    public void lunchClass() throws InterruptedException {
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--remote-allow-origins=*");
+
+        driver = new ChromeDriver(options);
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+        navigateToURL();
+    }
+   @AfterMethod
+   public void closeBrowser() {
+        driver.quit();
+   }
+
+
+    public void navigateToURL() throws InterruptedException {
+        driver.get(url);
+        Thread.sleep(2000);
+
+    }
+
+    public void provideEmail(String email) throws InterruptedException {
+        WebElement emailField = driver.findElement(By.cssSelector("input[type='email']"));
+        emailField.clear();
+        emailField.sendKeys(email);
+        Thread.sleep(2000);
+    }
+    public void providePassword(String password) throws InterruptedException {
+        WebElement passwordField = driver.findElement(By.cssSelector("input[type='password']"));
+        passwordField.clear();
+        passwordField.sendKeys(password);
+        Thread.sleep(2000);
+
+    }
+
+
+    public void searchForSong(String songName) throws InterruptedException {
+        WebElement searchField = driver.findElement(By.cssSelector("input[type='search']"));
+        searchField.clear();
+        searchField.sendKeys(songName);
+        WebElement viewAllField = driver.findElement(By.cssSelector("button[data-test='view-all-songs-btn']"));
+        viewAllField.click();
+        Thread.sleep(2000);
+
+    }
+
+    public void clickOnFirstSongToAdd() throws InterruptedException {
+        WebElement firstSong = driver.findElement(By.cssSelector("#songResultsWrapper > div > div > div.item-container > table > tr > td.title"));
+        firstSong.click();
+        WebElement addTo = driver.findElement(By.cssSelector(".btn-add-to"));
+        addTo.click();
+        Thread.sleep(2000);
+    }
+
+
+    public void clickSubmit() throws InterruptedException {
+        WebElement submit = driver.findElement(By.cssSelector("button[type='submit']"));
+        submit.click();
+        Thread.sleep(2000);
+
+    }
+    public void createPlaylist() {
+        WebElement newPlaylistName = driver.findElement(By.cssSelector("#songResultsWrapper .add-to input[data-test='new-playlist-name']"));
```

Comment:
//Comment from HW team
Hello,
I have reviewed your homework.
Great job on creating your Homework 17 and implementing TestNG

Result test PASSED
![image](https://github.com/user-attachments/assets/869e5f80-42e7-473b-9282-d5d6fbad8029)

But I have a tip for you when defining selector Avoid Absolute Paths, need short and clear. Your selector newPlaylistName long -> can define it as like: WebElement firstSong = `driver.findElement(By.cssSelector("#songResultsWrapper [class*=song-item]"));`